src/lib.rs

```diff
@@ -37,9 +37,11 @@ pub fn init_tracing_with_fallbacks(
         .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
         .with_writer(std::io::stdout.with_max_level(log_level));
 
+    let level_filter =
+        LevelFilter::from_level(std::cmp::max(log_level, tracing::Level::INFO));
     let subscriber = tracing_subscriber::registry()
         .with(fmt_layer)
-        .with(LevelFilter::INFO)
+        .with(level_filter)
```

Comment:
The thing is if isn't >=INFO the otel doesn't work.

Reply:
But that is what I'm doing with `max`

Reply:
```
    let level_filter =
        LevelFilter::from_level(std::cmp::max(log_level, tracing::Level::INFO));
```

Reply:
Is it clearer now?

```rust
    let level_filter: LevelFilter = max(log_level, tracing::Level::INFO).into();
```

Reply:
Tested my changes locally with stdout and jaeger and everything seems to work fine